bd-resilient-kv/AGENTS.md

```diff
@@ -31,14 +39,95 @@ The system provides efficient bulk operations through a consistent pattern:
 - Optimized for batch processing scenarios
 - Automatic timestamp synchronization for related entries
 
+### Versioned Storage Architecture
+
+The `VersionedKVStore` provides a higher-level API built on top of `VersionedKVJournal`:
+
+**Key Components**:
+- **VersionedKVJournal**: Low-level journal that tracks version numbers for each entry
+- **MemMappedVersionedKVJournal**: Memory-mapped persistence layer
+- **VersionedKVStore**: High-level HashMap-like API with automatic rotation and async write operations
+
+**Async API**:
+- Write operations (`insert()`, `remove()`, `rotate_journal()`) are async and require a Tokio runtime
+- Compression of archived journals is performed asynchronously using streaming I/O
+- Read operations remain synchronous and operate on the in-memory cache
+- The async API enables efficient background compression without blocking the main thread
+
+**Version Tracking**:
+- Every write operation (`insert`, `remove`) returns a monotonically increasing version number
+- Version numbers start at 1 (base version), first write is version 2
+- Entries with `Value::Null` are treated as deletions but still versioned
+
+**Rotation Strategy**:
+- Automatic rotation when journal size exceeds high water mark (triggered during async write operations)
+- Current state is compacted into a new journal as versioned entries
+- Old journal is archived with `.v{version}.zz` suffix
+- Archived journals are automatically compressed using zlib (RFC 1950, level 3) asynchronously
```

Comment:
FYI we use level 5 in the rest of the mobile code.

Reply:
Make sense to keep it consistent, I'll update